vm_test.go

```diff
@@ -153,6 +163,50 @@ func BenchmarkFibonnaci(b *testing.B) {
 	}
 }
 
+// Test for failures where both the callee and caller are making a tailcall.
+//
+// Related issue(s): #46
+func TestTailCall_Recursive(t *testing.T) {
+	s := `function tailcall(n, m)
+			if n > m then return n end
+			return tailcall(n + 1, m)
+		end
+		return tailcall(0, 5)`
+	testNoPanicString(t, s)
+}
+
+// Test for failures where only the caller is making a tailcall.
+//
+// Related issue(s): #46
+func TestTailCall_RecursiveDiffFn(t *testing.T) {
+	s := `function tailcall(n) return n+1 end
+		return tailcall(5)`
+	testNoPanicString(t, s)
+}
+
+// Test for failures where only the callee is making a tailcall.
+//
+// Related issue(s): #46
+func TestTailCall_SameFn(t *testing.T) {
```

Comment:
I'll just remove it since it's kind of redundant (the commit messages refer to it already). I do generally like knowing what prompted a test's creation, but if you don't think it's useful, that's fine.